keras/src/backend/openvino/trainer.py

```diff
@@ -0,0 +1,272 @@
+        elif isinstance(data, int):
+            param = ov_opset.parameter(shape=[], dtype=ov.Type.i32)
+            parametrize_data = OpenVINOKerasTensor(param.output(0))
+        elif isinstance(data, float):
+            param = ov_opset.parameter(shape=[], dtype=ov.Type.f32)
+            parametrize_data = OpenVINOKerasTensor(param.output(0))
+        else:
+            raise "Unknown type of input data {}".format(type(data))
+        return parametrize_data
+
+    def _get_compiled_model(self, data):
+        if (
+            self.ov_compiled_model is not None
+            and get_device() == self.ov_device
+        ):
+            return self.ov_compiled_model
+
+        # remove the previous cached compiled model if exists
+        del self.ov_compiled_model
+
+        # prepare parameterized input
+        self.struct_params = self._parameterize_data(data)
+        # construct OpenVINO graph during calling Keras Model
+        self.struct_outputs = self(self.struct_params)
+
+        parameters = []
+        for p in tree.flatten(self.struct_params):
+            parameters.append(p.output.get_node())
+        results = []
+        for r in tree.flatten(self.struct_outputs):
+            results.append(ov_opset.result(r.output))
+
+        # prepare compiled model from scratch
+        ov_model = ov.Model(results=results, parameters=parameters)
+        self.ov_compiled_model = ov.compile_model(ov_model, get_device())
+        self.ov_device = get_device()
+        return self.ov_compiled_model
+
+    def make_predict_function(self, force=False):
+        if self.predict_function is not None and not force:
+            return self.predict_function
+
+        def one_predict_step(data):
+            data = data[0]
+            return self.predict_step(data)
+
+        def multi_predict_steps(data):
+            outputs = one_predict_step(data[:1])
+
+            for single_step_data in data[1:]:
+                step_outputs = one_predict_step([single_step_data])
+                outputs = tree.map_structure(
+                    lambda t1, t2: np.concatenate([t1, t2]),
+                    outputs,
+                    step_outputs,
+                )
+            return outputs
+
+        if self.steps_per_execution > 1:
+            predict_step = multi_predict_steps
+        else:
+            predict_step = one_predict_step
+
+        self.predict_function = predict_step
+
+    def fit(
+        self,
+        x=None,
+        y=None,
+        batch_size=None,
+        epochs=1,
+        verbose="auto",
+        callbacks=None,
+        validation_split=0.0,
+        validation_data=None,
+        shuffle=True,
+        class_weight=None,
+        sample_weight=None,
+        initial_epoch=0,
+        steps_per_epoch=None,
+        validation_steps=None,
+        validation_batch_size=None,
+        validation_freq=1,
+    ):
+        raise NotImplementedError(
+            "`fit` is not supported with openvino backend"
```

Comment:
This seems like an important thing to support, can you attach a TODO with an issue number to this line?

Reply:
Hi @jeffcarp, `openvino` backend is only inference backend like numpy one. So no `fit` method is planned to support.
The plan is user can train his model using PT/JAX/TF and use openvino for inference right away without code change.

Reply:
Thanks for clarifying, I was not aware! Sounds good.